tests/rbd/test_rbd_image_zero_size.py

```diff
@@ -0,0 +1,115 @@
+"""Module to verify resize operation of RBD image with zero size.
+
+Test case covered -
+CEPH-83597243 -  Test to verify RBD image with zero size
+
+Pre-requisites :
+1. Cluster must be up and running with capacity to create pool
+2. We need atleast one client node with ceph-common package,
+   conf and keyring files
+
+Test Case Flow:
+1. Create a pool with name as 'images'
+2. Create an image inside this pool with 0 size' (rbd create images/image_zero1 --size 0)
+3. Create another image inside this pool with size 1024 (rbd create images/image_1024size --size 1024)
+4. Use rbd du command to check that image size for both images
+5. Resize second image size to 0
+6. Remove that images
+
+"""
+
+from ceph.parallel import parallel
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.rbd_utils import (
+    initial_rbd_config,
+    verify_image_feature_exist,
+    verify_image_size,
+)
+from ceph.rbd.workflows.rbd import resize_single_image_and_verify
+from utility.log import Log
+from utility.utils import run_fio
+from cli.rbd.rbd import Rbd
+import json
+
+log = Log(__name__)
+
+def resize_image_with_zero_size(rbd, rbd1, pool_type, **kw):
+    """
+        Test to verify RBD image with zero size.
+    Args:
+        rbd: RBD object
+        pool_type: pool type (ec_pool_config or rep_pool_config)
+        **kw: test data
+    """
+
+    try:
+        pool = kw["config"][pool_type]["pool"]
+        image = kw["config"][pool_type]["image"]
+        resize_to = kw["config"]["resize_to"]
+ 
+        if pool_type == "rep_pool_config": 
+            log.info(f"Resize Image {image} to zero \n\n")
+            resize_config = {"image-spec": f"{pool}/{image}", "size": resize_to, "allow-shrink": True}
+            out, err = rbd.resize(**resize_config)
+            if "100% complete" not in out + err:
+                log.error(
+                    f"Image {image} resize to 0 failed with error {out} {err}"
+                )
+                return 1
+
+        log.info(f"Verify Image {image} size")
+        info_spec = {"image-or-snap-spec": f"{pool}/{image}", "format": "json"}
+        out, err = rbd.info(**info_spec)
+        if err:
+            log.info(f"Error while fetching info for image {image}")
+            return 1
+        log.info(f"Image info: {out}")
+
+        out_json = json.loads(out)
+        if "0" not in str(out_json["size"]):
+            log.error(f"Image size Verification failed for {image}: Image size should be zero")
+            return 1
+
+        return 0
+
+    except RbdBaseException as error:
+        log.error(error.message)
+        return 1
+
+    finally:
+        log.info("Cleanup")
+        if pool_type == "ec_pool_config":
```

Comment:
pool clean up needs to be done for both rep-pool and EC-pool, please use this cleanup from here https://github.com/red-hat-storage/cephci/blob/master/ceph/rbd/workflows/cleanup.py
also make sure irrespective of test pass or fail pool should be deleted.